src/node/gov/handlers/proposals.h

```diff
@@ -131,10 +133,14 @@ namespace ccf::gov::endpoints
       ccf::jsgov::ProposalInfo& proposal_info,
       const std::string& constitution)
     {
-      std::vector<std::pair<MemberId, bool>> votes;
-      std::optional<jsgov::Failure> failure = std::nullopt;
-      std::optional<ccf::jsgov::Votes> final_votes = std::nullopt;
-      std::optional<ccf::jsgov::VoteFailures> vote_failures = std::nullopt;
+      // Create some temporaries to store resolution progress. These are written
+      // to proposal_info, and the KV, only for completed (accepted) proposal.
+      ccf::jsgov::Votes votes = {};
+      ccf::jsgov::VoteFailures vote_failures = {};
+      std::optional<ccf::jsgov::Failure> failure = std::nullopt;
```

Comment:
Is failure ever set? I'm not sure about the split, it doesn't seem to simplify the code much and means we allocate the vector every time.

Reply:
I'd initially hoped to remove the temporaries entirely, and write directly the `proposal_info`. I found some problem with that and restored them, but I think we're still slightly more explicit about their purpose now - they're not fields that are going to construct a `ProposalInfoSummary`, they're values that aren't _quite_ ready to re-populate `proposal_info`.